### PR TITLE
Add targeted block components to PlayerLocationVariable

### PR DESCRIPTION
**Pull request by cswhite2000**
Adds a few new components to `player-location` variables.

```java
    HAS_TARGET,
    TARGET_X,
    TARGET_Y,
    TARGET_Z,
    PLACE_X,
    PLACE_Y,
    PLACE_Z,
```

not sure whether caching here is worth it.

### CODE REVIEW

File: core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java [hunk 1]
```diff
@@ -1,17 +1,35 @@
...
+import java.util.HashMap;
 import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.block.RayBlockIntersection;
+import tc.oc.pgm.util.nms.NMSHacks;
 import tc.oc.pgm.variables.VariableDefinition;
 
 public class PlayerLocationVariable extends AbstractVariable<MatchPlayer> {
 
   private final Component component;
+  private final HashMap<MatchPlayer, RayBlockIntersection> locationTargetMap;
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java]: i would not bother caching a map, just cache literally the latest intersection:
```
Location inputCache;
RayBlockIntersection resultCache;
```

then just:
```
private RayBlockIntersection(Player player) {
  if (Objects.equals(player, inputCache) return resultCache;
  inputCache = player.getLocation().clone();
  return resultCache = PLAYER_UTILS.getTargetedBlock(player);
}
```

**Review by Pablete1234 — COMMENTED** [review]
consider also adding `HAS_TARGET` with 1 or 0 based on if the raycast found anything or not. I'm unsure as to what getBlock() returns if there was nothing found, but if there's potential for a NPE you should instead return 0 or -1 or something.

File: core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java (hunk 1 above)
Comment (cswhite2000) [reply core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java]: that would be easier

File: core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java [hunk 2]
```diff
@@ -30,6 +44,13 @@ protected double getValueImpl(MatchPlayer player) {
       case VEL_X -> player.getBukkit().getVelocity().getX();
       case VEL_Y -> player.getBukkit().getVelocity().getY();
       case VEL_Z -> player.getBukkit().getVelocity().getZ();
+      case TARGET_X -> intersection(player).getBlock().getX();
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java]: These all still have potential for NPE if a map author forgets to check for HAS_TARGET = 1

Comment (cswhite2000) [reply core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java]: right

File: core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java [hunk 3]
```diff
@@ -1,19 +1,36 @@
...
 
+import java.util.function.ToDoubleFunction;
+import org.bukkit.Location;
 import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.block.RayBlockIntersection;
 import tc.oc.pgm.variables.VariableDefinition;
 
 public class PlayerLocationVariable extends AbstractVariable<MatchPlayer> {
 
+  private static final double NULL_VALUE = -1000000; // negative 1 million
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java]: Honestly this sounds problematic, if someone tries to start placing blocks 1 million blocks out it's probs gonna force those chunks to load.

just make -1 the default or smthg

File: core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java [hunk 4]
```diff
@@ -1,19 +1,36 @@
...
 
+import java.util.function.ToDoubleFunction;
+import org.bukkit.Location;
 import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.block.RayBlockIntersection;
 import tc.oc.pgm.variables.VariableDefinition;
 
 public class PlayerLocationVariable extends AbstractVariable<MatchPlayer> {
 
+  private static final double NULL_VALUE = -1; // negative 1 million
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java]:
```suggestion
  private static final double NULL_VALUE = -1;
```

**Review by Pablete1234 — APPROVED** [review]

**Pull request merged**